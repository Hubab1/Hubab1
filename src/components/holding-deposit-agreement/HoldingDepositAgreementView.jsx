import React, { Fragment } from 'react';
import styled from '@emotion/styled';
import PropTypes from 'prop-types';
import { H1, SpacedH3, Spacer } from 'assets/styles';
import ActionButton from 'components/common/ActionButton/ActionButton';
import { BackLink } from 'components/common/BackLink';


import contract from 'assets/images/contract.svg';
import {css} from 'emotion';

export const Img = styled.img`
    padding-top: 14px;
    height: 157px;
`;

export const applicationUnit = css`
  color: #454B57;
  font-size: 14px;
  line-height: 17px;
  text-align: center;
  padding-top: 10px;
`;

export const HoldingDepositAgreementView = ({profile, configuration, handleContinue, handleClickBack}) => {
    if (!profile) return null;

    const {
        unit,
    } = profile;
    const buildingName = configuration.community.building_name || configuration.community.normalized_street_address;
    const unitNumber = (!!unit && !!unit.unit_number) ? ` Unit ${unit.unit_number}` : '';

    return (
        <Fragment>
            <H1>Holding Deposit Agreement</H1>
            <SpacedH3>You&apos;ll have to agree to some terms to move forward with the application process.</SpacedH3>
            <Img src={contract}/>
            <div className={applicationUnit}>{buildingName}{unitNumber}</div>
            <Spacer height={40}/>
            <ActionButton onClick={handleContinue} marginTop={30} marginBottom={20}>View and Sign Agreement</ActionButton>
            <BackLink to={handleClickBack}/>
        </Fragment>
    );
};

HoldingDepositAgreementView.propTypes = {
    profile: PropTypes.object,
    configuration: PropTypes.object,
    handleContinue: PropTypes.func,
    handleClickBack: PropTypes.string,
};

export default HoldingDepositAgreementView;
