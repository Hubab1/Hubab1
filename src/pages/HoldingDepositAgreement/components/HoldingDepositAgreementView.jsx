import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import { css } from 'emotion';

import ActionButton from 'common-components/ActionButton/ActionButton';
import { BackLink } from 'common-components/BackLink/BackLink';
import { H1, SpacedH3, Spacer } from 'assets/styles';
import contract from 'assets/images/contract.svg';

export const Img = styled.img`
    padding-top: 14px;
    height: 157px;
`;

export const applicationUnit = css`
    color: #454b57;
    font-size: 14px;
    line-height: 17px;
    text-align: center;
    padding-top: 10px;
`;

export const HoldingDepositAgreementView = ({ profile, configuration, handleContinue, handleClickBack }) => {
    if (!profile) return null;

    const { unit } = profile;
    const buildingName = configuration.community.building_name || configuration.community.normalized_street_address;
    const unitNumber = !!unit && !!unit.unit_number ? ` Unit ${unit.unit_number}` : '';

    return (
        <>
            <H1>Holding Deposit Agreement</H1>
            <SpacedH3>You&apos;ll have to agree to some terms to move forward with the application process.</SpacedH3>
            <Img src={contract} />
            <div className={applicationUnit}>
                {buildingName}
                {unitNumber}
            </div>
            <Spacer height={40} />
            <ActionButton onClick={handleContinue} marginTop={30} marginBottom={20}>
                View and Sign Agreement
            </ActionButton>
            <BackLink to={handleClickBack} />
        </>
    );
};

HoldingDepositAgreementView.propTypes = {
    profile: PropTypes.object,
    configuration: PropTypes.object,
    handleContinue: PropTypes.func,
    handleClickBack: PropTypes.string,
};

export default HoldingDepositAgreementView;
