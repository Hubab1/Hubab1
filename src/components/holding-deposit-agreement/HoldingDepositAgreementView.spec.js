import React from 'react';
import { shallow } from 'enzyme';
import HoldingDepositAgreementView from "./HoldingDepositAgreementView";


let defaultProps;

beforeEach(() => {
    defaultProps = {
        handleContinue: jest.fn(),
        handleClickBack: 'some_route',
        configuration: {
            community: {
                buildingName: 'Monterey Pines Apartments',

            }
        },
        profile: {
            unit: {
                unit_number: 'Unit 14F',
            }
        }
    };
});

it('Matches Snapshot', async () => {
    let wrapper = shallow( <HoldingDepositAgreementView {...defaultProps}/> );
    expect(wrapper.debug()).toMatchSnapshot();
});