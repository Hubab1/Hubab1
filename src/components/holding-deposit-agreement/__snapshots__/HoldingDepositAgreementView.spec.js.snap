// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Snapshot 1`] = `
"<Fragment>
  <Styled(h1)>
    Holding Deposit Agreement
  </Styled(h1)>
  <Styled(h3)>
    You&#39;ll have to agree to some terms to move forward with the application process.
  </Styled(h3)>
  <Styled(img) src=\\"contract.svg\\" />
  <div className=\\"css-lygmaw\\">
     Unit Unit 14F
  </div>
  <Styled(div) height={40} />
  <ActionButton onClick={[Function: mockConstructor]} marginTop={30} marginBottom={20} variant=\\"contained\\" color=\\"primary\\" successGreen={false}>
    View and Sign Agreement
  </ActionButton>
  <BackLink to={[Function: mockConstructor]} />
</Fragment>"
`;
